File: src/test/java/org/vanautrui/languages/compiler/parsing/astnodes/terminal/BoolConstNodeTest.java
package org.vanautrui.languages.compiler.parsing.astnodes.terminal;

import org.junit.Assert;
import org.junit.Test;
import org.vanautrui.languages.compiler.lexing.utils.CharacterList;
import org.vanautrui.languages.compiler.lexing.utils.TokenList;
import org.vanautrui.languages.compiler.lexing.tokens.BoolConstantToken;

public class BoolConstNodeTest {

    @Test
    public void test_parse_bool_constant_node() throws Exception {
        TokenList list = new TokenList();
        list.add(new BoolConstantToken(new CharacterList("true")));

        BoolConstNode node = new BoolConstNode(list);
        Assert.assertEquals(true,node.value);
    }
}

File: dragon-parser/src/test/astnodes/terminal/BoolConstNodeTest.cpp
package org.vanautrui.languages.test.astnodes.terminal;

import org.vanautrui.languages.compiler.lexing.tokens.BoolConstantToken;
import org.vanautrui.languages.compiler.lexing.utils.TokenList;
import org.vanautrui.languages.compiler.parsing.astnodes.terminal.BoolConstNode;

import java.nio.file.Paths;

public final class BoolConstNodeTest {

	public static int test_parse_bool_constant_node() throws Exception {
		final TokenList list = new TokenList(Paths.get("/dev/null"));
		list.add(new BoolConstantToken(true));

		final BoolConstNode node = new BoolConstNode(list);
		boolean assert1 = ( node.boolValue);

		return (assert1)?1:0;
	}
}
